src/recovery/store/sqlite.rs

```diff
@@ -425,7 +425,10 @@ impl SqliteProgressReader {
                 tx.send(kchange).await.unwrap();
             }
 
-            let sql = "SELECT execution, worker_index, frontier FROM progress";
+            // TODO: We really could only select progress from the
+            // latest execution.
+            let sql =
+                "SELECT execution, worker_index, frontier FROM progress ORDER BY execution ASC";
```

Comment:
Why don't we?

Reply:
Because I was tired yesterday. But I am not today. Done below.